### PR TITLE
bases the path on the current route to allow route params

### PR DESCRIPTION
**Pull request by maxgaj**
When setting the NgWizardComponent under a route with parameter, it will not work as the parameter wildcard will be used as a string.

to reproduce:

- set routing as
` { path: 'resource/:id/wizard', component: NgWizardComponent, children: [...] }`
- navigate to /resource/1/wizard/step1
- go to next step

Expected: new route is /resource/**1**/wizard/step2
Actual: new route is /resource/**:id**/wizard/step2

this PR proposes to solve this issue by basing the path on the current route

### CODE REVIEW

**Comment by maxgaj** [comment]
Added changes to:
- Allow multiple wizards in the same project by providing a 'name' option. Having unnamed wizard is still possible (for backward compatibility) but you can only have one of them
- Setting the NgWizardComponent inside nested routes. 'getWizardRoute' is now looking recursively inside every children route for the NgWizardComponent

File: projects/ng-wizard/src/lib/ng-wizard.service.ts [hunk 1]
```diff
@@ -132,9 +133,30 @@ export class NgWizardService {
...
+   * @param wizardName The unique name of the wizard
    */
-  private getWizardRoute(wizardComponentName: string): Route {
-    return this.router.config.find((route) => route.component && route.component.name === wizardComponentName);
+  private getWizardRoute(wizardComponentName: string, wizardName: string): Route {
+    const wizardRoutes = this.getAllWizardRoutes(this.router.config, wizardComponentName);
+    if (wizardName) {
+      return wizardRoutes.find((route) => route.data && route.data.name === wizardName);
+    } else {
+      return wizardRoutes.find((route) => !route.data || !route.data.name);
```
Comment (ibrahim-cm) [inline projects/ng-wizard/src/lib/ng-wizard.service.ts:143]: just a cosmetic change: the value being returned does not need to be in the 'else'

File: projects/ng-wizard/src/lib/ng-wizard.service.ts [hunk 2]
```diff
@@ -77,15 +78,15 @@ export class NgWizardService {
...
     const routeFragment = this.router.url.split('/');
-    routeFragment[routeFragment.length - 1] = stepData.path;
+    routeFragment.pop();
+    routeFragment.push(stepData.path);
     const stepPath = routeFragment.join('/');
 
     if (stepData.options.cleanQueryParameters) {
       return this.router.navigate([stepPath], { queryParams: {} });
     } else {
       return this.router.navigate([stepPath], { queryParamsHandling: 'merge' });
```
Comment (ibrahim-cm) [inline projects/ng-wizard/src/lib/ng-wizard.service.ts:88]: just a cosmetic change: the value being returned does not need to be in the 'else'

**Review by ibrahim-cm — APPROVED** [review]
for me it seems to be ok

**Pull request merged**